app/controllers/users_controller.rb

```diff
@@ -1,5 +1,10 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:id])
+    @posts = @user.posts
+  end
 end
```

Comment:
Well done implementing the suggestion by the previous reviewer, 😊

According to the project requirements, Clicking on each user takes you to their show page as shown in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png)

However, I encountered errors while trying to load users `show.html.erb`
![image](https://github.com/DianaBeki/Blog-app/assets/78251927/e3aacca9-ea20-4e6f-9071-d822b129f0aa)

I noticed that you have not properly set up [associations between models](https://guides.rubyonrails.org/v3.2.2/association_basics.html). We need associations between models because they make common operations simpler and easier in your code. an association is a connection between two Active Record models.

Types of associations:
```
belongs_to
has_one
has_many
... etc
```

Your code in `app/models/user.rb`
```ruby
class User < ApplicationRecord
  validates :name, presence: true
  validates :posts_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }

  def recent_posts
  ...
end
```

We are supposed to set up a `has_many` association with `posts`
```ruby
class User < ApplicationRecord
  has_many :posts
  validates :name, presence: true
  validates :posts_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }

  def recent_posts
  ...
end
```
- A `has_many` association indicates a one-to-many connection with another model, this indicate a user connection with posts.

This particular setup should have been completed in a previous milestone [Set up associations between models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md). Please check that to make all necessary corrections